metricbeat/module/elasticsearch/index_recovery/index_recovery.go

```diff
@@ -42,15 +42,17 @@ type MetricSet struct {
 func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 	config := struct {
 		ActiveOnly bool `config:"index_recovery.active_only"`
+		XPack      bool `config:"xpack.enabled"`
 	}{
 		ActiveOnly: true,
+		XPack:      false,
 	}
 	if err := base.Module().UnpackConfig(&config); err != nil {
 		return nil, err
 	}
 
 	localRecoveryPath := recoveryPath
-	if config.ActiveOnly {
+	if !config.XPack && config.ActiveOnly {
```

Comment:
See https://github.com/elastic/beats/blame/7.16/metricbeat/module/elasticsearch/index_recovery/index_recovery.go#L55

Reply:
I think that unless something is specifically broken, we shouldn't need to add this check back.

Reply:
Are we fine flipping the `ActiveOnly` default to `false` then ?

Reply:
Yes because the `xpack` flag will eventually dissapear. But users must still explicitly check to `true` in the case of `active_only`.